solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java

```diff
@@ -598,12 +599,48 @@ protected JWTAuthenticationResponse authenticate(String authorizationHeader) {
             } else {
               // Pull roles from separate claim, either as whitespace separated list or as JSON
               // array
-              Object rolesObj = jwtClaims.getClaimValue(rolesClaim);
+              Object rolesObj;
+              if (rolesClaim.indexOf('.') > 0) {
```

Comment:
Note that if someone uses `.` in a claim name, e.g. `my.roles` then we won't be able to configure it, since this code will look for `"my":{"roles":[]}` instead. Shold we first attempt fething the verbatim claim, and if not found, interpret it as nested? My gut feeling is that it is not that common to dot-separate claim names, it's more common with dash or underscore separators...

Reply:
Interesting point. I don't think that dots are common in claims. Maybe we could support, that a dot prepended with a backslash means the literal dot instead of as a child resolver. Such that `my.roles` is ` "my":{"roles":[]}` and `my\.roles` is `my.roles` - this could be done rather easily and also easily be added to the documentation.

Reply:
I pushed a commit that *first* checks the plain key name. If that claim is `null`, then we inspect nested... Please have a look.

Reply:
Looks reasonable to me. Should be clearly pointed out in documentation.

Reply:
The case where an access token contains both a `foo.bar` and a `foo: { bar }` entry is so narrow that I think it blurs more than it clarifies to add it to the docs. I hope this behaviour follows the least surprise rule. You are free to suggest re-wording of the docs though..

Reply:
What about changing 
_Both top-level claim and nested claim is supported. Use `someClaim.child` syntax to address a claim `child` nested within the `someClaim` object._
to 
_Both top-level claim and nested claim is supported. Use `someClaim.child` syntax to address the literal claim `someClaim.child` or, if it does not exist, the claim `child` nested within the `someClaim` object._